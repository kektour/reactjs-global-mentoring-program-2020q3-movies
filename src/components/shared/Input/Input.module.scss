.root {}

.title {
  text-transform: uppercase;
  color: #f65261;
  margin: 0;
  font-size: 14px;
  font-weight: bold;
  margin-bottom: 15px;
}

.input {
  height: 60px;
  border: none;
  padding: 20px;
  font-size: 18px;
  background-color: #555555;
  color: #fff;
  border-radius: 5px;
  width: 100%;

  &:focus {
    border: 1px solid #424242;
    outline: none;
  }
}
