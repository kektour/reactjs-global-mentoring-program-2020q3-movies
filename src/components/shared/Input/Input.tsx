import classnames from 'classnames';
import React from 'react';
import styles from './Input.module.scss';

type Props = {
  classes?: {
    root?: string;
  };
  title: string;
  nativeInputProps: Omit<React.InputHTMLAttributes<HTMLInputElement>, 'className'>;
};

export const Input: React.FC<Props> = (props) => {
  const { classes = {}, title, nativeInputProps } = props;
  return (
    <div className={classnames(styles.root, classes.root)}>
      <p className={styles.title}>{title}</p>
      <input className={styles.input} {...nativeInputProps} />
    </div>
  );
};
