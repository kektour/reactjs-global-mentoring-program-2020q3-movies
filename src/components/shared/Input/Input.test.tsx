import React from 'react';
import { render } from '@testing-library/react';
import { Input } from './Input';

describe('Input', () => {
  it('Should render component correctly', () => {
    const component = render(<Input title="Foo" nativeInputProps={{ defaultValue: 'Bar' }} />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
