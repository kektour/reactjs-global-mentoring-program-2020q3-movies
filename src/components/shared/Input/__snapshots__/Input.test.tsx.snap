// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <p
      class="title"
    >
      Foo
    </p>
    <input
      class="input"
      value="Bar"
    />
  </div>
</DocumentFragment>
`;
